### ISSUE TITLE
Unable to query entity with List of Map

### ISSUE THREAD

**Issue by yeefan** [mongodb/morphia#718]
Consider the following entity class:

```java
@Entity
public static class Foo {
    @Id
    public String id;

    // A list of free-form objects.
    public List<Map<String, Object>> data;

    @Override
    public String toString() {
        return ("id: " + id + ", data: " + data);
    }
}
```
And the following program fragment:

```java
MongoClient client = new MongoClient();
Morphia morphia = new Morphia();
morphia.map(Foo.class);
Datastore datastore = morphia.createDatastore(client, "test_foo");
datastore.ensureIndexes();

{
    Foo foo = new Foo();
    foo.id = "1";
    foo.data = Collections.singletonList(Collections.singletonMap("bar", "baz"));
    datastore.save(foo);
    System.out.println("Saved foo: " + foo);
}

{
    Query<Foo> query = datastore.find(Foo.class);
    Foo foo = query.get();  // Throws exception.
    System.out.println("Queried foo: " + foo);
}
```

The entity will be successfully and correctly added to the database, but the query will fail with the following exception:

```
Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: org.mongodb.morphia.mapping.MappingException: Embedded element isn't a DBObject! How can it be that is a class java.lang.String
	at org.mongodb.morphia.mapping.EmbeddedMapper.fromDBObject(EmbeddedMapper.java:171)
	at org.mongodb.morphia.mapping.Mapper.readMappedField(Mapper.java:617)
	at org.mongodb.morphia.mapping.Mapper.fromDb(Mapper.java:592)
	at org.mongodb.morphia.mapping.Mapper.fromDBObject(Mapper.java:306)
	at org.mongodb.morphia.query.MorphiaIterator.convertItem(MorphiaIterator.java:79)
	at org.mongodb.morphia.query.MorphiaIterator.processItem(MorphiaIterator.java:65)
	at org.mongodb.morphia.query.MorphiaIterator.next(MorphiaIterator.java:60)
	at org.mongodb.morphia.query.QueryImpl.get(QueryImpl.java:402)
	at Test2.main(Test2.java:44)
Caused by: java.lang.RuntimeException: org.mongodb.morphia.mapping.MappingException: Embedded element isn't a DBObject! How can it be that is a class java.lang.String
	at org.mongodb.morphia.mapping.EmbeddedMapper.fromDBObject(EmbeddedMapper.java:171)
	at org.mongodb.morphia.mapping.Mapper.readMappedField(Mapper.java:613)
	at org.mongodb.morphia.mapping.Mapper.fromDb(Mapper.java:570)
	at org.mongodb.morphia.mapping.EmbeddedMapper.readMapOrCollectionOrEntity(EmbeddedMapper.java:265)
	at org.mongodb.morphia.mapping.EmbeddedMapper.readCollection(EmbeddedMapper.java:207)
	at org.mongodb.morphia.mapping.EmbeddedMapper.fromDBObject(EmbeddedMapper.java:142)
	... 8 more
Caused by: org.mongodb.morphia.mapping.MappingException: Embedded element isn't a DBObject! How can it be that is a class java.lang.String
	at org.mongodb.morphia.mapping.EmbeddedMapper$1.eval(EmbeddedMapper.java:246)
	at org.mongodb.morphia.utils.IterHelper.loopMap(IterHelper.java:37)
	at org.mongodb.morphia.mapping.EmbeddedMapper.readMap(EmbeddedMapper.java:232)
	at org.mongodb.morphia.mapping.EmbeddedMapper.fromDBObject(EmbeddedMapper.java:140)
	... 13 more
```

Tried with Morphia 0.110 as well as 0.111-SNAPSHOT found at:
https://jenkins.10gen.com/job/morphia-build/

I suspect this might be related to issue #681, "List of List doesn't work".

**Comment by cteyton** [comment]
Simple check : what if you add an "@Embedded" annotation on "data" ?

**Comment by yeefan** [comment]
Doesn't work, same error message.

**Comment by yeefan** [comment]
I noticed the fix for issue #740 is almost the same as the issue I am raising here.

So I took the latest snapshot build from Jenkins (which I presume is build 218) and tried it out. Instead I get the following exception:

```
Exception in thread "main" java.lang.RuntimeException: java.lang.ClassCastException: java.lang.Class cannot be cast to java.lang.reflect.ParameterizedType
	at org.mongodb.morphia.mapping.EmbeddedMapper.fromDBObject(EmbeddedMapper.java:171)
	at org.mongodb.morphia.mapping.Mapper.readMappedField(Mapper.java:617)
	at org.mongodb.morphia.mapping.Mapper.fromDb(Mapper.java:592)
	at org.mongodb.morphia.mapping.Mapper.fromDBObject(Mapper.java:306)
	at org.mongodb.morphia.query.MorphiaIterator.convertItem(MorphiaIterator.java:79)
	at org.mongodb.morphia.query.MorphiaIterator.processItem(MorphiaIterator.java:65)
	at org.mongodb.morphia.query.MorphiaIterator.next(MorphiaIterator.java:60)
	at org.mongodb.morphia.query.QueryImpl.get(QueryImpl.java:402)
	at Test2.main(Test2.java:44)
Caused by: java.lang.ClassCastException: java.lang.Class cannot be cast to java.lang.reflect.ParameterizedType
	at org.mongodb.morphia.mapping.EmbeddedMapper.readCollection(EmbeddedMapper.java:194)
	at org.mongodb.morphia.mapping.EmbeddedMapper.fromDBObject(EmbeddedMapper.java:142)
	... 8 more
```

On the other hand, when I added the ```@Property``` annotation as given in the StackOverflow question (https://stackoverflow.com/questions/29032548/fetching-document-with-a-listmap-type-field-from-collection-in-mongodb-using-m), the program fragment worked.

I tried to read the wiki page (https://github.com/mongodb/morphia/wiki/PropertyAnnotation) but I don't understand what it does when no name argument is supplied to the ```@Property``` annotation.

So two questions:
1. Why is the ```@Property``` annotation required?
2. What does the ```@Property``` annotation do when no name argument is specified?

**Comment by evanchooly** [comment]
@Property *shouldn't* be required.  If there's no argument specified, the field name is used as the name to store in the document.  I'll see if I can get to the bottom of this one soon.

**Comment by yeefan** [comment]
I see, in other words ```@Property``` alone without any arguments should be a no-op. Thanks.

**Comment by yeefan** [comment]
I edited on my previous comment (https://github.com/mongodb/morphia/issues/718#issuecomment-85137583) to correct an error and also to expand on my observations. Sorry about that!

**Comment by yeefan** [comment]
Seems to be fixed in 0.111. Thanks!
